src/Option.ts

```diff
@@ -1117,98 +1103,6 @@ export const exists: {
   <A>(self: Option<A>, predicate: Predicate<A>): boolean => isNone(self) ? false : predicate(self.value)
 )
 
-// -------------------------------------------------------------------------------------
-// math
-// -------------------------------------------------------------------------------------
-
-/**
- * @category math
- * @since 2.0.0
- */
-export const sum: {
-  (self: Option<number>, that: Option<number>): Option<number>
-  (that: Option<number>): (self: Option<number>) => Option<number>
-} = lift2(N.sum)
-
-/**
- * @category math
- * @since 2.0.0
- */
-export const multiply: {
-  (self: Option<number>, that: Option<number>): Option<number>
-  (that: Option<number>): (self: Option<number>) => Option<number>
-} = lift2(N.multiply)
-
-/**
- * @category math
- * @since 2.0.0
- */
-export const subtract: {
-  (self: Option<number>, that: Option<number>): Option<number>
-  (that: Option<number>): (self: Option<number>) => Option<number>
-} = lift2(N.subtract)
-
-/**
- * @category math
- * @since 2.0.0
- */
-export const divide: {
-  (self: Option<number>, that: Option<number>): Option<number>
-  (that: Option<number>): (self: Option<number>) => Option<number>
-} = lift2(N.divide)
-
-/**
- * Sum all numbers in an iterable of `Option<number>` ignoring the `None` values.
- *
- * @param self - The iterable of `Option<number>` to be summed.
- *
- * @example
- * import { sumCompact, some, none } from 'effect/Option'
- *
- * const iterable = [some(2), none(), some(3), none()]
- * assert.deepStrictEqual(sumCompact(iterable), 5)
- *
- * @category math
- * @since 2.0.0
- */
-export const sumCompact = (self: Iterable<Option<number>>): number => {
-  let out = 0
-  for (const oa of self) {
-    if (isSome(oa)) {
-      out += oa.value
-    }
-  }
-  return out
-}
-
-/**
- * Multiply all numbers in an iterable of `Option<number>` ignoring the `None` values.
- *
- * @param self - The iterable of `Option<number>` to be multiplied.
- *
- * @example
- * import { multiplyCompact, some, none } from 'effect/Option'
- *
- * const iterable = [some(2), none(), some(3), none()]
- * assert.deepStrictEqual(multiplyCompact(iterable), 6)
- *
- * @category math
- * @since 2.0.0
- */
-export const multiplyCompact = (self: Iterable<Option<number>>): number => {
-  let out = 1
-  for (const oa of self) {
-    if (isSome(oa)) {
-      const a: number = oa.value
-      if (a === 0) {
-        return 0
-      }
-      out *= a
-    }
-  }
-  return out
-}
-
```

Comment:
We might want to keep these ... thoughts?

Reply:
Why do you want to remove lift2?

Reply:
multiplyCompact is the only fn here that can't be easily implemented as an alias of another fn. `sumCompact` is easily implemented in terms of `reduceCompact` but multiplyCompact has a performance improvement vs using `reduceCompact` because it ends computation when it finds a 0. I think for that reason it should be kept. Otherwise, the rest are somewhat useless.

Reply:
We can look at adding it back to the Number module if someone feels it is missing.